src/main/java/core/basesyntax/strategy/BalanceHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitTransaction;
+import java.util.List;
+
+public class BalanceHandler implements OperationHandler {
+    private static final String BALANCE = "b";
+
+    @Override
+    public List<FruitTransaction> getFruitFinalQuantity(List<FruitTransaction> listOfAll,
+                                                        List<FruitTransaction> listOfFruits) {
+        for (FruitTransaction listOfFruit : listOfFruits) {
+            int oldQuantity = 0;
+            for (FruitTransaction fruitTransaction : listOfAll) {
+                if (listOfFruit.getFruit().equals(fruitTransaction.getFruit())
+                        && fruitTransaction.getType().equals(BALANCE)) {
+                    int quantity = fruitTransaction.getQuantity();
+                    listOfFruit.setQuantity(quantity + oldQuantity);
+                    oldQuantity = quantity;
+                }
+            }
+        }
```

Comment:
The logic inside the method seems to be incorrect. It's currently iterating over the 'listOfFruits' and for each fruit, it iterates again over 'listOfAll' to find a matching fruit and update its quantity. This is not only inefficient but also does not seem to correctly apply the 'balance' operation as intended. You should consider revising the logic to correctly implement the balance operation according to the business requirements.